File: Task1/Task1.java
import java.util.Scanner;

public class Task1 {

    public static void FiboFunction(int n) {
        System.out.println("------------function based approach------------");
        
        long a = 0, b = 1;
        long startTime = System.nanoTime();
        long firstTerm = 0, secondTerm = 1;
        
        System.out.print("Series: ");
        for (int i = 2; i <= n; i++) {
            System.out.print(a + " , ");

            long next = a + b;
            a = b;
            b = next;
        }
        System.out.println(); 

        long endTime = System.nanoTime();
        long duration = (endTime - startTime); 

        System.out.println("Execution Time: " + duration + " nanoseconds");
    }

    static class fibo {
        
        public void exec(int n) {
            System.out.println("-------------------class based approach-------------------");
            
            long a = 0, b = 1;
            long startTime = System.nanoTime();

            long firstTerm = 0, secondTerm = 1;

            System.out.print("Series: ");
            for (int i = 2; i <= n; i++) {
                System.out.print(a + " , ");

                long next = a + b;
                a = b;
                b = next;
            }
            System.out.println();

            long endTime = System.nanoTime();
            long duration = (endTime - startTime);

            System.out.println("Execution Time: " + duration + " nanoseconds");
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter the num for fibbonaci : ");
        int count = sc.nextInt();

        //call function
        FiboFunction(count);

        //call class
        fibo cal = new fibo();
        cal.exec(count);

        sc.close();
    }
}

File: Task1/Task1.js
const readline = require('readline');
const { performance } = require('perf_hooks'); 

// 1. Function based approach
function fiboFunction(n) {
    console.log("------------function based approach------------");
    
    let a = 0, b = 1;

    let startTime = performance.now(); 
    process.stdout.write("Series: "); 
    
    for (let i = 2; i <= n; i++) {
        process.stdout.write(a + " , ");
        
        let next = a + b;
        a = b;
        b = next;
    }
    console.log(""); 

    let endTime = performance.now();
    let duration = (endTime - startTime).toFixed(4); 

    console.log("Execution Time: " + duration + " milliseconds");
}

// 2. Class based approach
class Fibo {
    exec(n) {
        console.log("-------------------class based approach-------------------");
        
        let a = 0, b = 1;
        let startTime = performance.now();

        process.stdout.write("Series: ");
        
        for (let i = 2; i <= n; i++) {
            process.stdout.write(a + " , ");
            
            let next = a + b;
            a = b;
            b = next;
        }
        console.log("");

        let endTime = performance.now();
        let duration = (endTime - startTime).toFixed(4);

        console.log("Execution Time: " + duration + " milliseconds");
    }
}

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question("Enter the num : ", function(input) {
    let count = parseInt(input); 

    // Function call karyu
    fiboFunction(count);

    // Class no object banavi ne call karyu
    let cal = new Fibo();
    cal.exec(count);

    rl.close();
});